From 9ecf274974a126a6937e1e14278cf895dd329dae Mon Sep 17 00:00:00 2001
From: Gustaw Lewandowski <gustaw.lewandowski@intel.com>
Date: Tue, 30 Apr 2019 00:00:28 +0200
Subject: [PATCH 1178/1214] ASoC: Intel: Skylake: bxt: Bound power states for
 DSP cores

DSP core 1 is not disabled in case reach zero on usage counter.

Change-Id: Ia1bc21a5ddc0d4aaed0b55056a3fc04a93f49226
Tracked-On: OAM-83010
---
 sound/soc/intel/skylake/bxt-sst.c | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/sound/soc/intel/skylake/bxt-sst.c b/sound/soc/intel/skylake/bxt-sst.c
index ab2f402..1cf36fb 100644
--- a/sound/soc/intel/skylake/bxt-sst.c
+++ b/sound/soc/intel/skylake/bxt-sst.c
@@ -572,7 +572,13 @@ static int bxt_set_dsp_D3(struct sst_dsp *ctx, unsigned int core_id)
 	int ret;
 	struct skl_ipc_dxstate_info dx;
 	struct skl_sst *skl = ctx->thread_context;
-	unsigned int core_mask = SKL_DSP_CORE_MASK(core_id);
+	unsigned int core_mask = SKL_DSP_CORES_MASK(skl->cores.count);
+
+	if (core_id != SKL_DSP_CORE0_ID)
+	{
+		dev_dbg(ctx->dev, "ignore D3 for core_id:%d\n", core_id);
+		return 0;
+	}
 
 	dx.core_mask = core_mask;
 	dx.dx_mask = SKL_IPC_D3_MASK;
@@ -604,6 +610,7 @@ static int bxt_set_dsp_D3(struct sst_dsp *ctx, unsigned int core_id)
 		return ret;
 	}
 	skl->cores.state[core_id] = SKL_DSP_RESET;
+	skl->cores.state[1] = SKL_DSP_RESET;
 	ret = skl_notify_tplg_change(skl, SKL_TPLG_CHG_NOTIFY_DSP_D3);
 	if (ret < 0)
 		dev_warn(ctx->dev,
-- 
2.7.4

